proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,797 @@
+                "advisoryurl":"https://github.com/advisories/GHSA-v76m-f5cx-8rg4"
+            },
+...
+            ]
+        }
+      ]
+    },
+    {
+      "framework": "net5.0",
+      "topLevelPackages": [ 
+        {
+          "id": "DotNetNuke.Core",
+          "requestedVersion": "6.0.0",
+          "resolvedVersion": "6.0.0",
+          "vulnerabilities" : [
+            {
+                "severity":"High",
+                "advisoryurl":"https://github.com/advisories/GHSA-g8j6-m4p7-5rfq"
+            },
+            {
+                "severity":"Moderate",
+                "advisoryurl":"https://github.com/advisories/GHSA-v76m-f5cx-8rg4"
+            },
+...
+            ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+### dotnet list package --include-transitive
+
+```dotnetcli
+Project 'MyProjectA' has the following package references
+   [netcoreapp3.1]:
+   Top-level Package                      Requested             Resolved
+   > Microsoft.Extensions.Primitives      [1.0.0, 5.0.0]        1.0.0
+   > NuGet.Commands                       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib                         [1.1.2, 2.0.0)        1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+...
+
+Project 'MyProjectB' has the following package references
+   [netcoreapp3.1]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+   > Microsoft.Win32.Primitives                                                         4.3.0
+...
+
+   [net5.0]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+...
+
+```
+
+### dotnet list package --include-transitive --json
```

Comment:
I would say its a bad user experience to have different defaults depending on if you specify `--format json` or not.  I do agree fewer options is simpler, but there's no `--do-not-include-transitive` option to opt out of it right?

Reply:
I think we should support *all* the same combinations we support today and the json sohuld output whatever the `--format` less command would. 

The only thing that I would recommend is that the json output schema is one and the same regardless of the option.

Reply:
The output command should simply display output for the context of the command. i.e. `--include-transitive --format json` would show them.

Reply:
@NuGet/nuget-client 
Currently this question main stumbling block. Whether we do `--include-transitive` implicitly for json output or not?
Please give your opinion. 
I don't have any strong opinion, I feel adding `--include-transitive` not really issue, I want to merge this PR soon.

Reply:
I do not believe you should include this implicitly. It defeats the purpose of the parameter itself even if the previous design was not ideal for user experience. We should instead for the existing experience in another issue than to try to fix it here.

Reply:
From offline discussion I strongly agree with `The behavior of the command is separate from the json output.` 
We should address it in separately, I don't find example of other application/tool with different defaults depending on output type, it set wrong example, hard to justify later.
I created separate tracking issue: https://github.com/NuGet/Home/issues/11550

Reply:
> I do not believe you should include this implicitly.

> The behavior of the command is separate from the json output.

I agree with these statements mainly because of this: `"parameters": "--include-transitive",`.
If there was no way to know by looking at this JSON file whether transitives were being included in the query, I'd ask for it to be made clear.

Reply:
> > I do not believe you should include this implicitly.
> 
> > The behavior of the command is separate from the json output.
> 
> I agree with these statements mainly because of this: `"parameters": "--include-transitive",`. If there was no way to know by looking at this JSON file whether transitives were being included in the query, I'd ask for it to be made clear.

We already have tracking issue for make `--include-transitive` by default, also I don't want to add `--include-transitive not included` for this, it's responsibility of user how tool works, check documentation before use.